Amplify/Categories/DataStore/Model/Temporal/TimeZone+Extension.swift

```diff
@@ -0,0 +1,136 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+
+import Foundation
+
+extension TimeZone {
+    private static let iso8601TimeZoneHHColonMMColonSSRegex = try? NSRegularExpression(pattern: "^[+-]\\d{2}:\\d{2}:\\d{2}$")
```

Comment:
From current `amplify-js` [implementation](https://github.com/aws-amplify/amplify-js/blob/main/packages/datastore/src/util.ts#L657), it only supports HH:MM:SS format.

Reply:
that's surprising, I ran a few live tests

HH:MM:SS works as expected
```
mutation MyMutationHHColonMMColonSS {
  createTest(input: {userID: "123", dt: "2023-11-30T11:04:03+08:00:30"}) {
    dt
  }
}
{
  "data": {
    "createTest": {
      "dt": "2023-11-30T11:04:03+08:00:30"
    }
  }
}
```

HHMM does not work
```
mutation MyMutationHHMM {
  createTest(input: {userID: "123", dt: "2023-11-30T11:04:03+0800"}) {
    dt
  }
}

{
  "data": null,
  "errors": [
    {
      "path": null,
      "locations": [
        {
          "line": 2,
          "column": 14,
          "sourceName": null
        }
      ],
      "message": "Validation error of type WrongType: argument 'input.dt' with value 'StringValue{value='2023-11-30T11:04:03+0800'}' is not a valid 'AWSDateTime' @ 'createTest'"
    }
  ]
}

```
So expectedly, HHMMSS also does not work..
```
mutation MyMutationHHMMSS {
  createTest(input: {userID: "123", dt: "2023-11-30T11:04:03+080030"}) {
    dt
  }
}
{
  "data": null,
  "errors": [
    {
      "path": null,
      "locations": [
        {
          "line": 2,
          "column": 14,
          "sourceName": null
        }
      ],
      "message": "Validation error of type WrongType: argument 'input.dt' with value 'StringValue{value='2023-11-30T11:04:03+080030'}' is not a valid 'AWSDateTime' @ 'createTest'"
    }
  ]
}
```

HH:MM:SS works as expected
```
mutation MyMutationHHColonMMColonSS {
  createTest(input: {userID: "123", dt: "2023-11-30T11:04:03+08:00:30"}) {
    dt
  }
}

{
  "data": {
    "createTest": {
      "dt": "2023-11-30T11:04:03+08:00:30"
    }
  }
}
```

Reply:
So what we have now looks good to me. 
- Data that is AWSDateTime with HH:MM:SS will be decoded properly. 

- Data would never be stored in the format HHMM since AppSync doesn't allow it, so even though we support decoding HHMM, that code path won't ever be used. This means it may pose the inverse problem: AppSync type is more restrictive than the Amplify Swift type. We allow storing date times with offset containing HHMM but the mutation with AppSync will fail. I believe this is still the correct thing to do since it conforms to the iSO 8601 spec and we can have varying types of data sources, ie. SQLite / local-only use cases.

- HHMMSS is not supported by AppSync AWSDateTime and not part of the ISO8601 spec.

Reply:
Thanks for verifying this with AppSync. 
We won't send date with string format `HHMM` or `HH:MM:SS`. We always send date to AppSync with string format`HH:MM` [code](https://github.com/aws-amplify/amplify-swift/blob/main/Amplify/Categories/DataStore/Model/Temporal/TemporalFormat.swift#L55). 

The format `HHMM` or `HH:MM:SS` are only the acceptable inputs. The code here is ensure we are able to decode these ISO8601 date formats to correct Tempral.DateTime instance. When encoding them back to string and submit to AppSync. They will always be in `HH:MM` format.